server/src/main/java/org/elasticsearch/cluster/metadata/MetadataUpdateSettingsService.java

```diff
@@ -67,11 +67,26 @@ public MetadataUpdateSettingsService(
         ThreadPool threadPool
     ) {
         this.clusterService = clusterService;
-        this.threadPool = threadPool;
         this.allocationService = allocationService;
         this.indexScopedSettings = indexScopedSettings;
         this.indicesService = indicesService;
         this.shardLimitValidator = shardLimitValidator;
+        this.threadPool = threadPool;
+        this.executor = (currentState, tasks) -> {
+            ClusterTasksResult.Builder<AckedClusterStateUpdateTask> builder = ClusterTasksResult.builder();
+            ClusterState state = currentState;
+            for (AckedClusterStateUpdateTask task : tasks) {
+                try {
+                    state = task.execute(state);
+                    builder.success(task);
+                } catch (Exception e) {
+                    builder.failure(task, e);
+                }
+            }
+            // reroute in case things change that require it (like number of replicas)
+            state = allocationService.reroute(state, "settings update");
```

Comment:
let's skip this in case the batch was a NOOP and `currentState == state`

Reply:
👍, be4de6c